pkg/collector/runner/runner_test.go

```diff
@@ -176,3 +176,9 @@ func TestStopCheck(t *testing.T) {
 	err = r.StopCheck(c2.ID())
 	assert.Equal(t, "timeout during stop operation on check id TestCheck:2", err.Error())
 }
+
+func TestGetCheckStats(t *testing.T) {
```

Comment:
I don't think that this test ensures that the structures are clones, only that the expected output is correct. Any way you can add tests for immutability of the underlying structures?

Reply:
They're not immutable :)

But I could check that they fail `==`.